### PR TITLE
circular-snout: pull live homepage and pupdates data from cms.glitch.me

### PR DESCRIPTION
**Pull request by taravancil**
## Links
* [circular-snout.glitch.me](https://circular-snout.glitch.me)

## Changes:
This pulls live homepage and pupdates data from cms.glitch.me instead of from the ~community filesystem.

- Remove `src/curated/{page.json}`
- Remove part of merge script that handles live updates to home.json
- Remove routes `/api/{pupdates|home}` and the helpers in `server/curated` that wrote pupdates.json and home.json to the filesystem

## How To Test:
* The homepage should load with all the content we expect
* Open pupdates, they should all load

## Feedback I'm looking for:
- How do we make sure cms.glitch.me is always online? Is it reasonable to assume that ~community will ping it for data often enough that it will never go to sleep? That feels risky.
- Does this approach seem sound?

### CODE REVIEW

File: server/curated.js [hunk 1]
```diff
@@ -1,36 +1,13 @@
...
-const readFile = util.promisify(fs.readFile);
-const writeFile = util.promisify(fs.writeFile);
-
 let pageCache = {};
 
 async function getData(page) {
   if (!pageCache[page]) {
-    const json = await readFile(path.join(__dirname, `../src/curated/${page}.json`));
-    pageCache[page] = JSON.parse(json);
+    const { data } = await axios.get(`https://cms.glitch.me/${page}.json`);
```
Comment (modernserf) [inline server/curated.js]: Where is this cache being updated?

File: sh/merge.sh [hunk 2]
```diff
@@ -8,10 +8,6 @@ if [ "$1" == "-h" ]; then
   exit 0
 fi
 
-curl https://community.glitch.me/api/home > src/curated/home.json
```
Comment (modernserf) [inline sh/merge.sh]: Does this script need to exist at all if not for this curated content?

**Comment by modernserf** [comment]
For preventing it from going to sleep, i imagine we could just add it to the Performance Boost team?

**Comment by oliviacpu** [comment]
@modernserf Does Performance Boost also keep awake, or just add more memory/resources?

File: sh/merge.sh (hunk 2 above)
Comment (oliviacpu) [reply sh/merge.sh]: I think a version of it needs to exist for getting live changes onto a branch now, but not directly pushed to master. Once we're not doing "Glitch hosted" Community anymore then we won't have "live changes" and probably won't need this.

Comment (modernserf) [reply sh/merge.sh]: what live changes are we doing besides CMS stuff?

File: server/curated.js (hunk 1 above)
Comment (oliviacpu) [reply server/curated.js]: I think there's another place we can put this that I'm going to try out

File: sh/merge.sh (hunk 2 above)
Comment (oliviacpu) [reply sh/merge.sh]: I think small fixes to deployed PRs have been considered fine "live changes" outside of the CMS stuff?

**Comment by modernserf** [comment]
So now the homepage/pupdate cache will update every hour -- does that work for marketing? I know they get concerned when the 'zine takes a while to update.

**Comment by oliviacpu** [comment]
@modernserf There is a 15 minute cache for zine content that we could use as well. I can change it to use that one.

**Comment by keithk** [comment]
Yeah, can we try making this a 5m cache?

**Comment by oliviacpu** [comment]
🙋 We're going to let the CMS push updates to us, so this is getting closed.

**Pull request closed without merging**